sympy/printing/pycode.py

```diff
@@ -493,8 +493,8 @@ def _print_seq(self, seq):
 
     def _print_MatMul(self, expr):
         "Matrix multiplication printer"
-        if isinstance(S(expr.args[0]), (Number, Symbol)):
-            return '({0})'.format(').dot('.join([self._print(expr.args[1]), self._print(expr.args[0])]))
+        if isinstance(expr.args[0], (Symbol, Number, Mul, Add)):
```

Comment:
```
A = MatrixSymbol("A", 3, 3)
k=Symbol("k",3,3)
 f = lambdify(A, (2+k)*A) or f = lambdify(A, (2*k)*A)
```
In this type expr.args[0] is may be` Mul` or `Add` type  so i am checking this also.

Reply:
All Symbol, Number will always be at position `0 `of expr.args.

Reply:
The scalar part of a MatMul can be any Expr type.

Reply:
You should use `expr.as_coeff_matrices()`